tracee-rules/output.go

```diff
@@ -44,24 +53,59 @@ func setupOutput(resultWriter io.Writer, clock Clock, webhook string) (chan type
 			}
 
 			if webhook != "" {
-				payload, err := prepareJSONPayload(res)
-				if err != nil {
-					log.Printf("error preparing json payload for %v: %v", res, err)
-					continue
+				if err := sendToWebhook(t, res, webhook, webhookTemplate, contentType, realClock{}); err != nil {
+					log.Println(err)
 				}
-				resp, err := http.Post(webhook, "application/json", strings.NewReader(payload))
-				if err != nil {
-					log.Printf("error calling webhook for %v: %v", res, err)
-					continue
-				}
-				resp.Body.Close()
 			}
 		}
-	}()
+	}(t)
 	return out, nil
 }
 
-func prepareJSONPayload(res types.Finding) (string, error) {
+func setupTemplate(webhookTemplate string, clock Clock) (*template.Template, error) {
+	return template.New(filepath.Base(webhookTemplate)).
+		Funcs(map[string]interface{}{
+			"timeNow": func(unixTs float64) string {
+				return clock.Now().UTC().Format("2006-01-02T15:04:05Z")
+			},
+		}).ParseFiles(webhookTemplate)
+}
+
+func sendToWebhook(t *template.Template, res types.Finding, webhook string, webhookTemplate string, contentType string, clock Clock) error {
+	var payload string
+
+	switch {
+	case webhookTemplate != "":
+		if t == nil {
+			return fmt.Errorf("error writing to template: template not initialized")
+		}
+		if contentType == "" {
+			return fmt.Errorf("--content-type flag needs to be defined for custom templates")
```

Comment:
not sure it should be required though (it's not required by HTTP)

Reply:
Yeah I agree, I changed it here https://github.com/aquasecurity/tracee/pull/582/commits/35fa90dc65a7fc5b0a8b9e492456f61369b795fc